src/NLog/Internal/SingleCallContinuation.cs

```diff
@@ -74,18 +74,13 @@ public void Function(Exception exception)
                     throw;
                 }
 
-                if (LogManager.ThrowExceptions)
-                {
-                    throw;
-                }
-
                 ReportExceptionInHandler(ex);
```

Comment:
Mh... `MustBeRethrown()` above already logs `ex`, so `ex` will be logged here for the second time.  
I think this line must be moved to before `MustBeRethrown()`.